docs/workflow/README.md

```diff
@@ -16,10 +16,37 @@ The runtime repo can be built from a regular, non-admin command prompt. The repo
 
 For information about the different options available, supply the argument `-help|-h` when invoking the build script:
 ```
-libraries -h
+build -h
 ```
 On Unix, arguments can be passed in with a single `-` or double hyphen `--`.
 
+### Configurations
+
+Sometimes you want to build this repo in a mixture of configurations. First, a quick reminder of some concepts -- see the [glossary](../project/glossary.md) for more on these:
+
+* CoreCLR (often referred to as the runtime, most code under src/coreclr) -- this is the execution engine for managed code. It is written in C/C++. When built in a checked configuration, it is easier to debug into it, but it executes managed code more slowly - so slowly it will take a long time to run the managed code unit tests
+ 
+* CoreLib (also known as System.Private.CoreLib - code under src/coreclr/System.Private.CoreLib) -- this is the lowest level managed library. It has a special relationship with the runtime -- it must be in the matching configuration. If the runtime you are using was built in a checked configuration, this must be in a checked configuration
+
+* All other libraries (most code under src/libraries) -- the bulk of the libraries are oblivious to the configuration that CoreCLR/CoreLib were built in. Like most code they are most debuggable when built in a checked configuration, and, happily, they still run sufficiently fast in that configuration that it's acceptable for development work.
+
+So if you're working in CoreCLR proper, you likely want to build everything in the checked configuration. If you're working in most libraries, you probably want to use checked libraries with release CoreCLR and CoreLib. And if you're working in CoreLib - you probably want to try to get the job done with release CoreCLR and CoreLib, and fall back to checked if you need to.
+
+How do you achieve that mixed configuration state? You pass the `/p:CoreCLRConfiguration` flag to the build, like so for Linux:
```

Comment:
```suggestion
How do you achieve that mixed configuration state? You pass the `/p:CoreCLRConfiguration` flag to the build, like so for Linux.  This will build a release CoreCLR (and CoreLib) and debug libraries:
```